src/Page.js

```diff
@@ -227,25 +242,80 @@ function getClosestHeading($, headingsSelector, element) {
 }
 
 /**
- * Records headings and keywords inside rendered page into this.headings and this.keywords respectively
+ * Checks if page.frontMatter has a valid page navigation specifier
  */
-Page.prototype.collectHeadingsAndKeywords = function () {
-  this.headings = {}; // clear any heading data from previous build
-  const $ = cheerio.load(fs.readFileSync(this.resultPath));
-  this.collectHeadingsAndKeywordsInContent($(`#${CONTENT_WRAPPER_ID}`).html(), null, false);
+Page.prototype.isPageNavigationSpecifierValid = function () {
+  const { pageNav } = this.frontMatter;
+  return pageNav && (pageNav === 'default' || Number.isInteger(pageNav));
 };
 
 /**
- * Generates a heading selector based on the indexing level
- * @param headingIndexingLevel to generate
+ * Generates element selector for page navigation, depending on specifier in front matter
  */
-function generateHeadingSelector(headingIndexingLevel) {
-  let headingsSelector = 'h1';
-  for (let i = 2; i <= headingIndexingLevel; i += 1) {
-    headingsSelector += `, h${i}`;
+Page.prototype.generateElementSelectorForPageNav = function (pageNav) {
+  if (pageNav === 'default') {
+    // Use specified navigation level or default in this.headingIndexingLevel
+    return `${generateHeadingSelector(this.headingIndexingLevel)}, panel`;
+  } else if (Number.isInteger(pageNav)) {
+    return `${generateHeadingSelector(parseInt(pageNav, 10))}, panel`;
   }
-  return headingsSelector;
-}
+  // Not a valid specifier
+  return undefined;
+};
+
+/**
+ * Collect headings outside of models and panels
+ * @param content, html content of a page
+ */
+Page.prototype.collectNavigableHeadings = function (content) {
+  const { pageNav } = this.frontMatter;
+  const elementSelector = this.generateElementSelectorForPageNav(pageNav);
+  if (elementSelector === undefined) {
+    return;
+  }
+  const $ = cheerio.load(content);
+  $('modal').remove();
+  $(elementSelector).each((i, elem) => {
+    // Check if heading or panel is already inside an unexpanded panel
+    let insideUnexpandedPanel = false;
```

Comment:
`insideUnexpandedPanel` -> `isInsideUnexpandedPanel` (to make it clearer that the variable is a boolean).